TestModels/Aggregate/runtimes/rust/src/operation/get_aggregate/_get_aggregate_output.rs

```diff
@@ -0,0 +1,214 @@
+// Code generated by software.amazon.smithy.rust.codegen.smithy-rs. DO NOT EDIT.
+#[allow(missing_docs)] // documentation missing in model
+#[non_exhaustive]
+#[derive(::std::clone::Clone, ::std::cmp::PartialEq, ::std::fmt::Debug)]
+pub struct GetAggregateOutput {
+    #[allow(missing_docs)] // documentation missing in model
+    pub simple_string_list: ::std::option::Option<::std::vec::Vec<::std::string::String>>,
+    #[allow(missing_docs)] // documentation missing in model
+    pub structure_list: ::std::option::Option<::std::vec::Vec<crate::types::StructureListElement>>,
+    #[allow(missing_docs)] // documentation missing in model
+    pub simple_string_map: ::std::option::Option<
+        ::std::collections::HashMap<::std::string::String, ::std::string::String>,
+    >,
+    #[allow(missing_docs)] // documentation missing in model
+    pub simple_integer_map:
+        ::std::option::Option<::std::collections::HashMap<::std::string::String, i32>>,
+    #[allow(missing_docs)] // documentation missing in model
+    pub nested_structure: ::std::option::Option<crate::types::NestedStructure>,
+}
+impl GetAggregateOutput {
+    #[allow(missing_docs)] // documentation missing in model
+    ///
+    /// If no value was sent for this field, a default will be set. If you want to determine if no value was sent, use `.simple_string_list.is_none()`.
```

Comment:
Interesting, I see that this is what `smithy-rs` does, but to me it seems just as dangerous as it is convenient. I think it's worth getting background on this decision from @rcoh

Reply:
I do see it's controlled by a `flattenCollectionAccessors` config setting that seems to be off by default.

Reply:
yeah—this is most useful for AWS models that essentially unchangeable and have incorrect modeled behavior (for example, a collection in return position modeled as optional)—it is very unlikely that you as a consumer cared about that distinction.

If you control the model, then this isn't necessary–because you should model it as required!

Reply:
RFC: https://github.com/smithy-lang/smithy-rs/blob/main/design/src/rfcs/rfc0035_collection_defaults.md

Reply:
Thanks for the background @rcoh! So sounds like we need to be prepared to adjust for this when wrapping a Rust AWS SDK into Dafny, but for our own local service models we should just go by whether the member is `@required` or not.

I also made a note on https://github.com/smithy-lang/smithy-dafny/issues/373 that we should add coverage of required aggregate members to the test model (after getting this merged)